### PR TITLE
Methods for finding the largest connected component and largest connected hypergraph

### PR DESCRIPTION
**Pull request by nwlandry**
* Added a method for finding the largest connected component of a hypergraph
* Added a method for finding the largest connected hypergraph from a dataset
* Added unit tests for these methods.
* All unit tests pass
* Formatted with black

### CODE REVIEW

**Review by leotrs — APPROVED** [review]
Awesome!

File: xgi/algorithms/connected.py [hunk 1]
```diff
@@ -114,6 +122,39 @@ def number_connected_components(H):
...
+
+    Parameters
+    ----------
+    H: Hypergraph object
+        The hypergraph of interest
+
+    Returns
+    -------
+    set
+        the largest connected component of the hypergraph.
```
Comment (maximelucas) [inline xgi/algorithms/connected.py:137]: Start with capital T. Also, explicitly say that this is a set of nodes? Contrary to largest_connected_hypergraph

File: xgi/algorithms/connected.py [hunk 2]
```diff
@@ -163,3 +204,45 @@ def _plain_bfs(H, source):
...
+    ----------
+    H: Hypergraph object
+        The hypergraph of interest
+    in_place: bool, optional
+        If False, creates a copy; if True, modifies the existing hypergraph
+
+    Returns
+    -------
+    None
+        if in_place (modifies the existing hypergrpah)
```
Comment (maximelucas) [inline xgi/algorithms/connected.py:223]: If

File: xgi/algorithms/connected.py [hunk 3]
```diff
@@ -163,3 +204,45 @@ def _plain_bfs(H, source):
...
+
+    xgi.Hypergraph
+       if not in_place: the hypergraph induced on the nodes of the
```
Comment (maximelucas) [inline xgi/algorithms/connected.py:226]: If

**Review by maximelucas — CHANGES_REQUESTED** [review]
Great work! Just the tiniest things

File: xgi/algorithms/connected.py (hunk 2 above)
Comment (nwlandry) [reply xgi/algorithms/connected.py:223]: Done.

**Review by maximelucas — APPROVED** [review]

**Pull request merged**